ios/brave-ios/Sources/Brave/BraveSkus/BraveSkusAccountLink.swift

```diff
@@ -0,0 +1,89 @@
+
+import Foundation
+import WebKit
+import os.log
+import Preferences
+import AIChat
+
+private extension BraveStoreProduct {
+  /// The key to use when storing the receipt in WebKit's LocalStorage
+  var localStorageKey: String {
+    switch self {
+    case .vpnMonthly, .vpnYearly: return "braveVpn.receipt"
+    case .leoMonthly, .leoYearly: return "braveLeo.receipt"
+    }
+  }
+}
+
+/// A class for linking Product Purchases with a Brave Account
+class BraveSkusAccountLink {
+  /// An enum representing the Skus Environment
+  private enum Environment: String, CaseIterable {
+    /// The skus development environment
+    case development
+    
+    /// The skus staging environment
+    case staging
+    
+    /// The skus production environment
+    case production
+    
+    /// The host/domain for the environment
+    var host: String {
+      switch self {
+      case .development: return "account.brave.softwware"
+      case .staging: return "account.bravesoftware.com"
+      case .production: return "account.brave.com"
+      }
+    }
+  }
+  
+  /// Injects Skus product order receipt information into WebKit's LocalStorage for use by the `Environment.host` page
+  /// - Parameter webView: The web-view whose LocalStorage to inject the product order information
+  /// - Parameter product: The product whose receipt information to inject
+  @MainActor
+  @discardableResult static func injectLocalStorage(webView: WKWebView, product: BraveStoreProduct) async -> Bool {
+    // The WebView has no URL so do nothing
+    guard let url = webView.url else {
+      return false
+    }
+    
+    // The URL must have a scheme and host
+    guard let components = URLComponents(url: url, resolvingAgainstBaseURL: false),
+          let scheme = components.scheme,
+          let host = components.host else {
+      return false
+    }
+
+    // Validate the scheme and host is the correct page to inject sensitive purchase information
+    // Purchase information must only be injected into a secure page and must only be injected into
+    // one of the Brave account pages as determined by `Environment.host`
+    if scheme != "https" || !Environment.allCases.map({ $0.host }).contains(host) {
+      return false
+    }
+    
+    do {
+      // Retrieve the LocalStorage Key and Receipt to inject
+      let storageKey = product.localStorageKey
+      let receipt = try BraveSkusSDK.receipt(for: product)
+      
+      // Inject the receipt into LocalStorage
+      try await webView.evaluateSafeJavaScriptThrowing(functionName: "localStorage.setItem", args: [storageKey, receipt], contentWorld: .defaultClient)
+      
+      // Brave-Leo requires Order-ID to be injected into LocalStorage.
+      if let orderId = Preferences.AIChat.subscriptionOrderId.value {
+        try await webView.evaluateSafeJavaScriptThrowing(functionName: "localStorage.setItem", args: ["braveLeo.orderId", orderId], contentWorld: .defaultClient)
```

Comment:
Yes. The reason I did defaultClient is so the page cannot use prototype pollution to mess with the LocalStorage APIs.

So we can write to the page and storage, but the page cannot read my code. It can read the LocalStorage receipt.